src/librustc/ty/sty.rs

```diff
@@ -1975,8 +1975,12 @@ impl<'tcx> Const<'tcx> {
         ty: ParamEnvAnd<'tcx, Ty<'tcx>>,
     ) -> &'tcx Self {
         let ty = tcx.lift_to_global(&ty).unwrap();
-        let layout = tcx.layout_of(ty).unwrap_or_else(|e| {
-            panic!("could not compute layout for {:?}: {:?}", ty, e)
+        let layout = tcx.layout_of(ty).unwrap_or_else(|_| {
+            // FIXME: add delay_span_bug call, we can only get here if there are errors
+            // we produce a weird dummy layout with somewhat sane values
+            let mut layout = tcx.layout_of(ParamEnv::reveal_all().and(tcx.types.u128)).unwrap();
+            layout.ty = ty.value;
+            layout
```

Comment:
Woah.

Your comment consists of two sentences, right, despite there not being a `.` nor the first word of the second sentence being capitalized? I am not sure but so far I haven't found another way to parse these sentences.  Please add punctuation :)

And also please add that `delay_span_bug`. I mean I have no clue when and where and how this method is used, but just using `u128` clearly cannot be right. Maybe add a comment saying why this even remotely makes sense -- i.e., is the result of this method unused in the error case, or so?

Reply:
(I read it as one sentence. "weird" and "dummy" are adjectives modifying "layout", and "with somewhat sane values" is also attached to "layout.")

Reply:
GitHub cut away half the comment. I cannot parse the following as one sentence:

> add delay_span_bug call, we can only get here if there are errors we produce a weird dummy layout with somewhat sane values